hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/EmbeddableMappingTypeImpl.java

```diff
@@ -447,12 +489,32 @@ private boolean finishInitialization(
 		// Reset the attribute mappings that were added in previous attempts
 		attributeMappings.clear();
 
+
 		for ( final Property bootPropertyDescriptor : bootDescriptor.getProperties() ) {
 			final AttributeMapping attributeMapping;
 
 			final Type subtype = subtypes[attributeIndex];
 			final Value value = bootPropertyDescriptor.getValue();
+			final OnExecutionGenerator onExecutionGenerator;
+			if ( onExecutionCompositeGenerator != null ) {
+				onExecutionGenerator = (OnExecutionGenerator) onExecutionCompositeGenerator
+						.getGenerator( attributeIndex );
+			}
+			else {
+				onExecutionGenerator = null;
+			}
+
 			if ( subtype instanceof BasicType ) {
+				final boolean insertable;
+				final boolean updateable;
+				if ( generatedWithNoParameter( onExecutionGenerator ) ) {
```

Comment:
Please help me understand, why would we want to set `insertable`/`updatable` of a  property to `false` when it has a generator. Doesn't this depend on the generation types? Or should we "care" about the config at all i.e. rely on the use configuring this correctly? Also, how does this interact with the `writable` flag of `@org.hibernate.annotations.Generated`?